lock/src/test/java/org/infinispan/lock/ClusteredLockSplitBrainTest.java

```diff
@@ -59,118 +79,132 @@ public void testLockUseAfterPartitionWithoutMajority() throws Throwable {
-               return null;
-            }));
-
-      assertFailureFromMinorityPartition(lock3);
+      assertFailureFromMinorityPartition(lock4);
+      assertFailureFromMinorityPartition(lock5);
    }
 
    @Test
    public void testAutoReleaseIfLockIsAcquiredFromAMinorityPartition() throws Throwable {
 
       ClusteredLockManager clm0 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(0));
       ClusteredLockManager clm1 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(1));
+      ClusteredLockManager clm2 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(2));
 
       assertTrue(clm0.defineLock(LOCK_NAME));
 
       ClusteredLock lock0 = clm0.get(LOCK_NAME);
       ClusteredLock lock1 = clm1.get(LOCK_NAME);
+      ClusteredLock lock2 = clm2.get(LOCK_NAME);
 
       await(lock0.tryLock());
       assertTrue(await(lock0.isLockedByMe()));
 
-      splitCluster(new int[]{0}, new int[]{1, 2, 3});
+      splitCluster(new int[]{0}, new int[]{1, 2, 3, 4, 5});
+
+      CompletableFuture<Boolean> tryLock1Result = lock1.tryLock(1, TimeUnit.SECONDS);
+      CompletableFuture<Boolean> tryLock2Result = lock2.tryLock(1, TimeUnit.SECONDS);
+
+      assertTrue("Just one of the locks has to work", await(tryLock1Result) ^ await(tryLock2Result));
 
       assertFailureFromMinorityPartition(lock0);
-      assertTrue(await(lock1.tryLock(1, TimeUnit.SECONDS)));
    }
 
-   private void assertFailureFromMinorityPartition(ClusteredLock lock3) {
-      await(lock3.lock()
-            .thenRun(() -> {
+   @Test
+   public void testTryLocksBeforeSplitBrain() throws Throwable {
+
+      ClusteredLockManager clm0 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(0));
+      ClusteredLockManager clm1 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(1));
+      ClusteredLockManager clm2 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(2));
+
+      assertTrue(clm0.defineLock(LOCK_NAME));
+
+      ClusteredLock lock0 = clm0.get(LOCK_NAME);
+      ClusteredLock lock1 = clm1.get(LOCK_NAME);
+      ClusteredLock lock2 = clm2.get(LOCK_NAME);
+
+      CompletableFuture<Boolean> tryLock1 = lock1.tryLock();
+      CompletableFuture<Boolean> tryLock2 = lock2.tryLock();
+
+      splitCluster(new int[]{0}, new int[]{1, 2, 3, 4, 5});
+
+      assertTrue("Just one of the locks has to work", await(tryLock1) ^ await(tryLock2));
+
+      assertFailureFromMinorityPartition(lock0);
+   }
+
+   private void assertTryLock(ClusteredLock lock) {
+      assertTrue("Lock acquisition should be true " + lock, await(lock.tryLock(29, TimeUnit.SECONDS)
+            .thenApply(tryLockRequest -> {
+               if (tryLockRequest) {
+                  await(lock.unlock());
+               }
+               return tryLockRequest;
+            }).exceptionally(ex -> {
+               fail("Should not be failing from majority partition " + lock + " " + ex.getMessage());
```

Comment:
I don't think `exceptionally()` is needed when you expect the call to succeed, `await()` will rethrow the exception anyway.

Reply:
I added this more in case it fails to have an explanation of where it failed. Does this deserve to be removed then?